### PR TITLE
Boomers Ruin The Economy - Pt. 1.5: Account Management

### PR DESCRIPTION
**Pull request by MarkSuckerberg**
## About The Pull Request
Adds management and creation of bank accounts through two new NTOS apps. One manages staff in your department's paychecks, the other allows for the creation, monitoring, and removal of bank accounts.

## Why It's Good For The Game
Bank accounts have long been an intangible and untouchable concept. This aims to remedy this by allowing department heads to give you pay raises and cuts, as well as giving the Captain, HoP, and QM (to a lesser degree, the QM can only create accounts, no removals or adjustments) the ability to add new accounts to wayward ashwalkers and charlie station scientists.  

## Changelog
:cl:
add: NTOS App for the management of all bank accounts
add: NTOS App for the management of departmental paychecks and termination
tweak: Allows the freezing/unfreezing of accounts and budgets from the appropriate management app
/:cl:

### CODE REVIEW

File: code/modules/economy/account.dm [hunk 1]
```diff
@@ -50,6 +54,8 @@
 	return FALSE
 
 /datum/bank_account/proc/payday(amt_of_paychecks, free = FALSE)
+	if(frozen)
+		bank_card_talk("ERROR: Payday aborted, account frozen!")
```
Comment (Maldaris) [inline code/modules/economy/account.dm:58]: there's no return method/break here. Does this actually abort the payout?

File: code/modules/economy/account.dm [hunk 2]
```diff
@@ -58,6 +64,8 @@
 	else
 		var/datum/bank_account/D = SSeconomy.get_dep_account(account_job.paycheck_department)
 		if(D)
```
Comment (Maldaris) [inline code/modules/economy/account.dm:66]: Same here, presumably needs an elseif?

File: code/modules/economy/account.dm (hunk 1 above)
Comment (MarkSuckerberg) [reply code/modules/economy/account.dm:58]: this explains a lot

File: code/modules/economy/account.dm [hunk 3]
```diff
@@ -5,6 +5,8 @@
 	var/account_balance = 0
 	var/datum/job/account_job
 	var/list/bank_cards = list()
+	///If the account has been frozen by someone at an account management terminal.
+	var/frozen = FALSE
```
Comment (Superyodeler) [inline code/modules/economy/account.dm:9]: When attempting to withdraw money from a frozen card I get the message "ERROR: The linked account requires -5022.22 more credits to perform that withdrawal." We need a message to make it clear what's actually happening.

**Review by Superyodeler — CHANGES_REQUESTED** [review]
Deleting an account doesn't work and there needs to be a message for attempting to withdraw funds from a frozen card. It's also possible to put two entries on the same account number, but I'm not sure that's a problem.

**Review by Superyodeler — APPROVED** [review]
Looks good!

**Comment by triplezeta** [comment]
Account creation works like a charm. Goes for the entire bank account creation app; it just works. Setting paycheck amounts is easy, creating new accounts is also easy.

The departmental account manager app is less reliable. Account deletion does not seem to work properly. Changing the paycheck using the departmental account manager causes it to null. Freezing budget cards works properly, preventing transactions in and out of it, but it also will cause it to eat any holochips put in.

**Comment by Superyodeler** [comment]
I think the departmental manager doesn't allow you to delete accounts, just demote to assistant

**Pull request merged**